### ISSUE TITLE
Deprecated constant which will be removed in HA Core 2025.1

### ISSUE THREAD

**Issue by robitsrl** [smartHomeHub/SmartIR#1186]
I'm having the 2 following logs entry:

Entity None (<class 'custom_components.smartir.climate.SmartIRClimate'>) is using deprecated supported features values which will be removed in HA Core 2025.1. Instead it should use <ClimateEntityFeature.TARGET_TEMPERATURE|FAN_MODE: 9>,

HVAC_MODE_FAN_ONLY was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.FAN_ONLY instead, please report it to the author of the 'smartir' custom integration
HVAC_MODE_AUTO was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.AUTO instead, please report it to the author of the 'smartir' custom integration
SUPPORT_TARGET_TEMPERATURE was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.TARGET_TEMPERATURE instead, please report it to the author of the 'smartir' custom integration
SUPPORT_FAN_MODE was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.FAN_MODE instead, please report it to the author of the 'smartir' custom integration
SUPPORT_SWING_MODE was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.SWING_MODE instead

 please report it to the author of the 'smartir' custom integration and reference https://developers.home-assistant.io/blog/2023/12/28/support-feature-magic-numbers-deprecation

**Comment by AlePezzo** [comment]
I have same issue:

Logger: homeassistant.components.climate.const
Source: helpers/deprecation.py:204
Integration: Climatizzatore (documentation, issues)
First occurred: 09:11:30 (9 occurrences)
Last logged: 09:11:30

HVAC_MODE_FAN_ONLY was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.FAN_ONLY instead, please report it to the author of the 'smartir' custom integration
HVAC_MODE_AUTO was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.AUTO instead, please report it to the author of the 'smartir' custom integration
SUPPORT_TARGET_TEMPERATURE was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.TARGET_TEMPERATURE instead, please report it to the author of the 'smartir' custom integration
SUPPORT_FAN_MODE was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.FAN_MODE instead, please report it to the author of the 'smartir' custom integration
SUPPORT_SWING_MODE was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.SWING_MODE instead, please report it to the author of the 'smartir' custom integration

**Comment by akacruel** [comment]
I can confirm the issue

**Comment by aIexus** [comment]
The same...

**Comment by Claudio1L** [comment]
same

**Comment by maxbaron** [comment]
same problem and same answer from Alexa

**Comment by Claudio1L** [comment]
also looking to the last activities, looks like not maintained so much. Anyone have been able to find a solution ?

**Comment by mesut3000** [comment]
Hi, I have the same situation too.

**Comment by schteff** [comment]
I can add the following for media players (same issue, just different enum)

```
Entity None (<class 'custom_components.smartir.media_player.SmartIRMediaPlayer'>) is using deprecated supported features values which will be removed in HA Core 2025.1. Instead it should use <MediaPlayerEntityFeature.VOLUME_MUTE|TURN_ON|TURN_OFF|PLAY_MEDIA|VOLUME_STEP|SELECT_SOURCE: 3976>
```

**Comment by felipecrs** [comment]
I think it's fine guys, no rush. This project has 1 full year to adapt to these changes. :)

**Comment by RunoFawkes** [comment]
> same, and for those climate entities also have problems with Alexa. Alexa telling there are issues but switch the climate on or off.

Is this Alexa behavior related to this issue?
I've been having the same problem, if I ask Alexa to either turn my thermostat on or off, she does it successfully but says "the thermostat is not responding" (temp changes are handled and responded to correctly).

I believe this has to do with an incomplete response being sent to Alexa, and I believe this was already fixed in the base Generic Thermostat in Home Assistant, [here](https://github.com/home-assistant/core/issues/15580)

Do I need to open a separate issue for that?
@felipecrs

**Comment by felipecrs** [comment]
I would say that's a different issue indeed. This deprecation warning was not supposed to change or affect any functionality.

Maybe the issue is even in Home Assistant, as I don't think this integration interacts with Alexa directly.

PS: I am not a maintainer, this is just my opinion.

**Comment by Claudio1L** [comment]
Updating core to 2024.1.3 Alexa issue was solved.  Still there the Deprecated logs, of course

**Comment by lollo78** [comment]
Same for me on climate:

`2024-01-23 05:05:39.809 WARNING (MainThread) [homeassistant.components.climate.const] HVAC_MODE_OFF was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.OFF instead, please report it to the author of the 'smartir' custom integration
2024-01-23 05:05:39.825 WARNING (MainThread) [homeassistant.components.climate.const] HVAC_MODE_HEAT was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.HEAT instead, please report it to the author of the 'smartir' custom integration
2024-01-23 05:05:39.839 WARNING (MainThread) [homeassistant.components.climate.const] HVAC_MODE_COOL was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.COOL instead, please report it to the author of the 'smartir' custom integration
2024-01-23 05:05:39.853 WARNING (MainThread) [homeassistant.components.climate.const] HVAC_MODE_DRY was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.DRY instead, please report it to the author of the 'smartir' custom integration
2024-01-23 05:05:39.866 WARNING (MainThread) [homeassistant.components.climate.const] HVAC_MODE_FAN_ONLY was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.FAN_ONLY instead, please report it to the author of the 'smartir' custom integration
2024-01-23 05:05:39.879 WARNING (MainThread) [homeassistant.components.climate.const] HVAC_MODE_AUTO was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.AUTO instead, please report it to the author of the 'smartir' custom integration
2024-01-23 05:05:39.893 WARNING (MainThread) [homeassistant.components.climate.const] SUPPORT_TARGET_TEMPERATURE was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.TARGET_TEMPERATURE instead, please report it to the author of the 'smartir' custom integration
2024-01-23 05:05:39.906 WARNING (MainThread) [homeassistant.components.climate.const] SUPPORT_FAN_MODE was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.FAN_MODE instead, please report it to the author of the 'smartir' custom integration
2024-01-23 05:05:39.921 WARNING (MainThread) [homeassistant.components.climate.const] SUPPORT_SWING_MODE was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.SWING_MODE instead, please report it to the author of the 'smartir' custom integration`

and also this:

`2024-01-23 05:06:06.592 WARNING (MainThread) [homeassistant.helpers.entity] Entity None (<class 'custom_components.smartir.climate.SmartIRClimate'>) is using deprecated supported features values which will be removed in HA Core 2025.1. Instead it should use <ClimateEntityFeature.TARGET_TEMPERATURE|FAN_MODE: 9>, please report it to the author of the 'smartir' custom integration and reference https://developers.home-assistant.io/blog/2023/12/28/support-feature-magic-numbers-deprecation
2024-01-23 05:06:06.598 WARNING (MainThread) [homeassistant.helpers.entity] Entity None (<class 'custom_components.smartir.media_player.SmartIRMediaPlayer'>) is using deprecated supported features values which will be removed in HA Core 2025.1. Instead it should use <MediaPlayerEntityFeature.VOLUME_MUTE|PREVIOUS_TRACK|NEXT_TRACK|TURN_ON|TURN_OFF|PLAY_MEDIA|VOLUME_STEP|SELECT_SOURCE: 4024>, please report it to the author of the 'smartir' custom integration and reference https://developers.home-assistant.io/blog/2023/12/28/support-feature-magic-numbers-deprecation`

**Comment by vacari** [comment]
Same here

**Comment by SrBurner** [comment]
Same Here, I'm adding some moore

SUPPORT_SET_SPEED was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use FanEntityFeature.SET_SPEED instead, please report it to the author of the 'smartir' custom integration
SUPPORT_DIRECTION was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use FanEntityFeature.DIRECTION instead, please report it to the author of the 'smartir' custom integration
SUPPORT_OSCILLATE was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use FanEntityFeature.OSCILLATE instead, please report it to the author of the 'smartir' custom integration
SUPPORT_SWING_MODE was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.SWING_MODE instead, please report it to the author of the 'smartir' custom integration
SUPPORT_FAN_MODE was used from fglair_heatpump_controller, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.FAN_MODE instead, please create a bug report at https://github.com/bigmoby/fglair_for_homeassistant/issues
SUPPORT_PRESET_MODE was used from fglair_heatpump_controller, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.PRESET_MODE instead, please create a bug report at https://github.com/bigmoby/fglair_for_homeassistant/issues
SUPPORT_SWING_MODE was used from fglair_heatpump_controller, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.SWING_MODE instead, please create a bug report at https://github.com/bigmoby/fglair_for_homeassistant/issues
SUPPORT_TARGET_TEMPERATURE was used from fglair_heatpump_controller, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.TARGET_TEMPERATURE instead, please create a bug report at https://github.com/bigmoby/fglair_for_homeassistant/issues
Entity None (<class 'custom_components.smartir.climate.SmartIRClimate'>) does not set ClimateEntityFeature.TURN_ON but implements the turn_on method. Please report it to the author of the 'smartir' custom integration
Entity None (<class 'custom_components.smartir.climate.SmartIRClimate'>) implements HVACMode(s): off, heat, cool, dry, fan_only, heat_cool and therefore implicitly supports the turn_on/turn_off methods without setting the proper ClimateEntityFeature. Please report it to the author of the 'smartir' custom integration
Entity None (<class 'custom_components.fglair_heatpump_controller.climate.FujitsuClimate'>) does not set ClimateEntityFeature.TURN_OFF but implements the turn_off method. Please create a bug report at https://github.com/bigmoby/fglair_for_homeassistant/issues
Entity None (<class 'custom_components.fglair_heatpump_controller.climate.FujitsuClimate'>) does not set ClimateEntityFeature.TURN_ON but implements the turn_on method. Please create a bug report at https://github.com/bigmoby/fglair_for_homeassistant/issues
Entity None (<class 'custom_components.fglair_heatpump_controller.climate.FujitsuClimate'>) implements HVACMode(s): off, heat, cool, auto, dry, fan_only and therefore implicitly supports the turn_on/turn_off methods without setting the proper ClimateEntityFeature. Please create a bug report at https://github.com/bigmoby/fglair_for_homeassistant/issues

**Comment by remontti** [comment]
here the integration stopped working! 

```
root@core-ssh:/config# cat home-assistant.log | grep smartir
2024-02-11 17:16:42.045 WARNING (SyncWorker_3) [homeassistant.loader] We found a custom integration smartir which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-02-11 17:17:23.751 WARNING (MainThread) [homeassistant.components.climate.const] HVAC_MODE_OFF was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.OFF instead, please report it to the author of the 'smartir' custom integration
2024-02-11 17:17:23.759 WARNING (MainThread) [homeassistant.components.climate.const] HVAC_MODE_HEAT was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.HEAT instead, please report it to the author of the 'smartir' custom integration
2024-02-11 17:17:23.767 WARNING (MainThread) [homeassistant.components.climate.const] HVAC_MODE_COOL was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.COOL instead, please report it to the author of the 'smartir' custom integration
2024-02-11 17:17:23.775 WARNING (MainThread) [homeassistant.components.climate.const] HVAC_MODE_DRY was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.DRY instead, please report it to the author of the 'smartir' custom integration
2024-02-11 17:17:23.782 WARNING (MainThread) [homeassistant.components.climate.const] HVAC_MODE_FAN_ONLY was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.FAN_ONLY instead, please report it to the author of the 'smartir' custom integration
2024-02-11 17:17:23.790 WARNING (MainThread) [homeassistant.components.climate.const] HVAC_MODE_AUTO was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.AUTO instead, please report it to the author of the 'smartir' custom integration
2024-02-11 17:17:23.797 WARNING (MainThread) [homeassistant.components.climate.const] SUPPORT_TARGET_TEMPERATURE was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.TARGET_TEMPERATURE instead, please report it to the author of the 'smartir' custom integration
2024-02-11 17:17:23.805 WARNING (MainThread) [homeassistant.components.climate.const] SUPPORT_FAN_MODE was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.FAN_MODE instead, please report it to the author of the 'smartir' custom integration
2024-02-11 17:17:23.812 WARNING (MainThread) [homeassistant.components.climate.const] SUPPORT_SWING_MODE was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.SWING_MODE instead, please report it to the author of the 'smartir' custom integration
2024-02-11 17:17:27.436 WARNING (MainThread) [homeassistant.components.fan] SUPPORT_SET_SPEED was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use FanEntityFeature.SET_SPEED instead, please report it to the author of the 'smartir' custom integration
2024-02-11 17:17:27.445 WARNING (MainThread) [homeassistant.components.fan] SUPPORT_DIRECTION was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use FanEntityFeature.DIRECTION instead, please report it to the author of the 'smartir' custom integration
2024-02-11 17:17:27.454 WARNING (MainThread) [homeassistant.components.fan] SUPPORT_OSCILLATE was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use FanEntityFeature.OSCILLATE instead, please report it to the author of the 'smartir' custom integration
2024-02-11 17:17:27.462 WARNING (MainThread) [homeassistant.components.fan] SUPPORT_SET_SPEED was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use FanEntityFeature.SET_SPEED instead, please report it to the author of the 'smartir' custom integration
2024-02-11 17:17:27.470 WARNING (MainThread) [homeassistant.components.fan] SUPPORT_DIRECTION was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use FanEntityFeature.DIRECTION instead, please report it to the author of the 'smartir' custom integration
2024-02-11 17:17:27.479 WARNING (MainThread) [homeassistant.components.fan] SUPPORT_OSCILLATE was used from smartir, this is a deprecated constant which will be removed in HA Core 2025.1. Use FanEntityFeature.OSCILLATE instead, please report it to the author of the 'smartir' custom integration
2024-02-11 17:17:40.980 WARNING (MainThread) [homeassistant.setup] Setup of smartir is taking over 10 seconds.
2024-02-11 17:18:33.420 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: smartir
2024-02-11 17:19:33.530 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: smartir
2024-02-11 17:20:33.613 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: smartir
2024-02-11 17:21:33.686 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: smartir
2024-02-11 17:22:24.228 ERROR (MainThread) [homeassistant.setup] Setup of 'smartir' is taking longer than 300 seconds. Startup will proceed without waiting any longer
2024-02-11 17:22:24.231 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform 'smartir.climate': Unable to set up component.
2024-02-11 17:22:24.232 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform 'smartir.climate': Unable to set up component.
2024-02-11 17:22:24.233 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform 'smartir.climate': Unable to set up component.
2024-02-11 17:22:24.233 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform 'smartir.climate': Unable to set up component.
2024-02-11 17:22:24.234 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform 'smartir.media_player': Unable to set up component.
2024-02-11 17:22:24.235 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform 'smartir.media_player': Unable to set up component.
2024-02-11 17:22:24.235 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform 'smartir.fan': Unable to set up component.

```

**Comment by felipecrs** [comment]
Maybe Python 3.12 incompatibility...

**Comment by PeteDenmark** [comment]
Same log entries. Hope it will get fixed before, otherwise my Broadlink is just a bad paper weight. Smart-ir was the reason i got the Broadlink.

**Comment by litinoveweedle** [comment]
Hi, I just pushed PR #1231 fixing all those warning and errors.

@vassilis-panos Coudl you please review it, merge it and release if suitable? It also seems, that you have many PR pending, mostly related to new added codes. Are you planning to merge those? If you need any help please let me know.

If anyone would like to meanwhile get rid of the errors, you can use my fork from here: https://github.com/litinoveweedle/SmartIR

**Comment by emanuelemattiolo** [comment]
I also warning for all HVAC_MODE* constants.  Thanks

**Comment by litinoveweedle** [comment]
> I also warning for all HVAC_MODE* constants. Thanks

No worries, we know, everyone is getting those. ;-)

**Comment by litinoveweedle** [comment]
> 2024-03-29 18:22:26.213 WARNING (MainThread) [homeassistant.helpers.entity] Entity None (<class 'custom_components.smartir.media_player.SmartIRMediaPlayer'>) is using deprecated supported features values which will be removed in HA Core 2025.1. Instead it should use <MediaPlayerEntityFeature.VOLUME_MUTE|PREVIOUS_TRACK|NEXT_TRACK|TURN_ON|TURN_OFF|PLAY_MEDIA|VOLUME_STEP|SELECT_SOURCE: 4024>, please report it to the author of the 'smartir' custom integration and reference https://developers.home-assistant.io/blog/2023/12/28/support-feature-magic-numbers-deprecation

Hey, thank for notice, I think I can prepare PR for this as well. Bu the problem seems to be the author willingness to merge it. But no worry, I can fork this repo and maintain it at least for the time being, as I am using it myself and I need it to work. ;-)

**Comment by litinoveweedle** [comment]
Hello, was fixed in the [release](https://github.com/litinoveweedle/SmartIR/releases/tag/1.17.10) of my forked version
